master-runner/src/main/scala/de/thm/ii/fbs/services/DockerService.scala

```diff
@@ -0,0 +1,86 @@
+package de.thm.ii.fbs.services
+
+import de.thm.ii.fbs.types.DockerCmdConfig
+
+import scala.sys.process.{Process, ProcessLogger}
+import scala.util.Properties
+
+/**
+  * Helper object for running Docker Containers
+  *
+  * @author Max Stephan
+  */
+object DockerService {
+  private val DOCKER_CMD = "docker"
+  private val DOCKER_RUN = "run"
+  private val DOCKER_REMOVE = "--rm"
+
+  private def buildCmd(config: DockerCmdConfig): Seq[String] = {
+    val builder = Seq.newBuilder[String]
+    builder += DOCKER_CMD
+    builder += DOCKER_RUN
+    builder += DOCKER_REMOVE
+    builder ++= config.mount.flatMap(getMountString)
+    builder ++= config.dockerOptions
+    builder += config.image
+    builder ++= config.runOptions
+
+    builder.result()
+  }
+
+  private def getMountString(mount: (String, String)): Seq[String] = {
+    Seq("-v", s"${mount._1}:${mount._2}")
+  }
+
+
+  /**
+    * Function to run a Docker Container
+    *
+    * @param config Config for the Docker Command
+    * @return (exit code, stdout, stderr)
+    */
+  def runContainer(config: DockerCmdConfig): (Int, String, String) = {
+    val cmd = buildCmd(config)
+
+    val stdoutStream = new StringBuilder
+    val stderrStream = new StringBuilder
+    val procLogger = ProcessLogger((o: String) => stdoutStream.append(o), (e: String) => stderrStream.append(e))
+
+    // Run Docker Container
+    val exitCode = Process(cmd).!(procLogger)
```

Comment:
- The process is not started asynchronously, which of course blocks the thread, is there a possibility to do it differently in Scala? (I have not found anything)